packages/dashboard/app/withdraw/utils/transactionService.ts

```diff
@@ -0,0 +1,277 @@
+
+      // Double gas for retry
+      txOptions.gasPrice = initialGasPrice.mul(2);
+
+      console.log('Retrying with increased gas price:', ethers.utils.formatUnits(txOptions.gasPrice, 'gwei') + ' gwei');
+
+      const retryTx = await pkpWallet.sendTransaction(txOptions);
+      const retryReceipt = await provider.waitForTransaction(retryTx.hash, 1);
+
+      return {
+        success: retryReceipt.status === 1,
+        hash: retryTx.hash,
+        receipt: retryReceipt
+      };
+    } else if (error.message && error.message.includes('insufficient funds for intrinsic transaction cost')) {
+      return {
+        success: false,
+        hash: '',
+        error: 'Insufficient funds for transaction. Please add more ETH to your wallet.'
+      };
+    } else {
+      return {
+        success: false,
+        hash: '',
+        error: error.message || 'Failed to send transaction'
+      };
+    }
+  }
+}
+
+/**
+ * Calculates the gas costs for ETH transactions on Base network
+ */
+export async function calculateEthGasCosts(): Promise<{
+  totalCost: ethers.BigNumber;
+  gasLimit: ethers.BigNumber;
+  gasPrice: ethers.BigNumber;
+}> {
+  const baseProvider = new ethers.providers.JsonRpcProvider(BASE_MAINNET_RPC);
+  const ethProvider = new ethers.providers.JsonRpcProvider(ETH_MAINNET_RPC);
+
+  // Get current gas prices for L1 and L2
+  const l1GasPrice = await ethProvider.getGasPrice();
+  const l2GasPrice = await baseProvider.getGasPrice();
+  
+  // Fixed gas limits
+  const l1GasLimit = ethers.BigNumber.from(1600);
+  const l2GasLimit = ethers.BigNumber.from(23100);
+  
+  // Calculate gas costs
+  const l1GasCost = l1GasLimit.mul(l1GasPrice);
+  const l2GasCost = l2GasLimit.mul(l2GasPrice);
+  const buffer = l1GasCost.add(l2GasCost).mul(50).div(100);
+  
+  // Total gas cost includes L1 fee, L2 gas, and buffer
+  const totalCost = l1GasCost.add(l2GasCost).add(buffer);
+  
+  const fixedBuffer = ethers.utils.parseEther('0.000001'); // 0.000001 ETH additional buffer
+  const finalCost = totalCost.add(fixedBuffer);
+  
+  return {
+    totalCost: finalCost,
+    gasLimit: l2GasLimit,
+    gasPrice: l2GasPrice
+  };
+}
+
+/**
+ * Sends an ETH transfer transaction
+ */
+export async function sendEthTransaction({
+  pkpWallet,
+  amount,
+  recipientAddress,
+  tokenBalance
+}: SendEthTransactionParams): Promise<TransactionResult> {
+  try {
+    const provider = new ethers.providers.JsonRpcProvider(BASE_MAINNET_RPC);
+
+    // Get gas info
+    const gasInfo = await calculateEthGasCosts();
+
+    // Check if amount is valid (the gas costs are already accounted for in handleMaxAmount)
+    if (amount.gt(tokenBalance)) {
+      return {
+        success: false,
+        hash: '',
+        error: `Insufficient balance for transaction.`
+      };
+    }
+
+    if (amount.lt(gasInfo.totalCost)) {
+      return {
+        success: false,
+        hash: '',
+        error: `Insufficient balance for gas fees.`
+      };
+    }
```

Comment:
```suggestion
    if (tokenBalance.lt(amount.add(gasInfo.totalCost))) {
      return {
        success: false,
        hash: '',
        error: `Insufficient balance to withdraw specified amount and pay for current gas fees.`
      };
    }
```

The second check is avoiding user to withdraw less than the gas they have to pay (might sound stupid but if they have that `tokenBalance` available after gas they should be able to do so)

Nonetheless, what you want to check is that with your balance you can pay for gas and retire the amount you want. Both things at the same time, together.
Having the checks separated might create edge cases where both conditions are true but the combination is not (ex. user wants to withdraw the 100% of their balance which is also more than the gas they will have to pay)